SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc

```diff
@@ -19,259 +19,385 @@
-    LogDebug("ME0ReDigiProducer")
-      << "ME0DetId " << detId << " central TOF " << centralTOF << std::endl;
-  }
-  nPartitions_ = centralTOF_.size()/6;
-  LogDebug("ME0ReDigiProducer")<<" Number of partitions "<<nPartitions_<<std::endl;
+	// set geometry
+	edm::ESHandle<ME0Geometry> hGeom;
+	eventSetup.get<MuonGeometryRecord>().get(hGeom);
+	geometry= &*hGeom;
+
+	if(useBuiltinGeo){
+		const auto& chambers = geometry->chambers();
+		if(!chambers.size())
+			throw cms::Exception("Setup") << "ME0ReDigiProducer::beginRun() - No ME0Chambers in geometry.";
+		const unsigned int nLayers = chambers.front()->nLayers();
+		if(nLayers != layerReadout.size() )
+			throw cms::Exception("Configuration") << "ME0ReDigiProducer::beginRun() - The geoemtry has "<<nLayers
+			<< " layers, but the readout of "<<layerReadout.size() << " were specified with the layerReadout parameter."  ;
+		fillCentralTOFs();
+	} else {
+		LogDebug("ME0ReDigiProducer")
+				<< "Building temporary geometry:" << std::endl;
+		tempGeo = new TemporaryGeometry(geometry,numberOfSrips,numberOfPartitions);
+		LogDebug("ME0ReDigiProducer")
+		<< "Done building temporary geometry!" << std::endl;
+
+		if(tempGeo->numLayers() != layerReadout.size() )
+			throw cms::Exception("Configuration") << "ME0ReDigiProducer::beginRun() - The geoemtry has "<<tempGeo->numLayers()
+			<< " layers, but the readout of "<<layerReadout.size() << " were specified with the layerReadout parameter."  ;
+	}
 }
 
 
 void ME0ReDigiProducer::produce(edm::Event& e, const edm::EventSetup& eventSetup)
 {
-  edm::Service<edm::RandomNumberGenerator> rng;
-  CLHEP::HepRandomEngine* engine = &rng->getEngine(e.streamID());
+	edm::Service<edm::RandomNumberGenerator> rng;
+	CLHEP::HepRandomEngine* engine = &rng->getEngine(e.streamID());
 
-  edm::Handle<ME0DigiPreRecoCollection> input_digis;
-  e.getByToken(token_, input_digis);
+	edm::Handle<ME0DigiPreRecoCollection> input_digis;
+	e.getByToken(token, input_digis);
 
-  std::unique_ptr<ME0DigiPreRecoCollection> output_digis(new ME0DigiPreRecoCollection());
+	std::unique_ptr<ME0DigiPreRecoCollection> output_digis(new ME0DigiPreRecoCollection());
+	std::unique_ptr<ME0DigiPreRecoMap>       output_digimap(new ME0DigiPreRecoMap());
 
-  // build the clusters
-  buildDigis(*(input_digis.product()), *output_digis, engine);
+	// build the digis
+	buildDigis(*(input_digis.product()),
+			*output_digis,
+			*output_digimap,
+			engine);
 
-  // store them in the event
-  e.put(std::move(output_digis));
+	// store them in the event
+	e.put(std::move(output_digis));
+	e.put(std::move(output_digimap));
 }
 
 
 void ME0ReDigiProducer::buildDigis(const ME0DigiPreRecoCollection & input_digis,
-                                   ME0DigiPreRecoCollection & output_digis,
-                                   CLHEP::HepRandomEngine* engine)
+		ME0DigiPreRecoCollection & output_digis,
+		ME0DigiPreRecoMap & output_digimap,
+		CLHEP::HepRandomEngine* engine)
 {
-  /*
```

Comment:
Are these deleted comments no longer correct? It would be nice to replace them with an updated version, if so.